Source/Fluids/WarpXFluidContainer.cpp

```diff
@@ -0,0 +1,1854 @@
+                    if (N_arr(i,j,k+1) > 0.0) {
+                        Ux_pz = (NUx_arr(i, j, k+1) / N_arr(i,j,k+1));
+                        Uy_pz = (NUy_arr(i, j, k+1) / N_arr(i,j,k+1));
+                        Uz_pz = (NUz_arr(i, j, k+1) / N_arr(i,j,k+1));
+                    }
+                    if (N_arr(i,j,k-1) > 0.0) {
+                        Ux_mz = (NUx_arr(i, j, k-1) / N_arr(i,j,k-1));
+                        Uy_mz = (NUy_arr(i, j, k-1) / N_arr(i,j,k-1));
+                        Uz_mz = (NUz_arr(i, j, k-1) / N_arr(i,j,k-1));
+                    }
+
+#elif defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+
+                    amrex::Real Ux_px = 0.0, Ux_mx = 0.0, Uy_px = 0.0, Uy_mx = 0.0, Uz_px = 0.0, Uz_mx = 0.0;
+                    if (N_arr(i+1,j,k) > 0.0) {
+                        Ux_px = (NUx_arr(i+1, j, k) / N_arr(i+1,j,k));
+                        Uy_px = (NUy_arr(i+1, j, k) / N_arr(i+1,j,k));
+                        Uz_px = (NUz_arr(i+1, j, k) / N_arr(i+1,j,k));
+                    }
+                    if (N_arr(i-1,j,k) > 0.0) {
+                        Ux_mx = (NUx_arr(i-1, j, k) / N_arr(i-1,j,k));
+                        Uy_mx = (NUy_arr(i-1, j, k) / N_arr(i-1,j,k));
+                        Uz_mx = (NUz_arr(i-1, j, k) / N_arr(i-1,j,k));
+                    }
+
+                    amrex::Real Ux_pz = 0.0, Ux_mz = 0.0, Uy_pz = 0.0, Uy_mz = 0.0, Uz_pz = 0.0, Uz_mz = 0.0;
+                    if (N_arr(i,j+1,k) > 0.0) {
+                        Ux_pz = (NUx_arr(i, j+1, k) / N_arr(i,j+1,k));
+                        Uy_pz = (NUy_arr(i, j+1, k) / N_arr(i,j+1,k));
+                        Uz_pz = (NUz_arr(i, j+1, k) / N_arr(i,j+1,k));
+                    }
+                    if (N_arr(i,j-1,k) > 0.0) {
+                        Ux_mz = (NUx_arr(i, j-1, k) / N_arr(i,j-1,k));
+                        Uy_mz = (NUy_arr(i, j-1, k) / N_arr(i,j-1,k));
+                        Uz_mz = (NUz_arr(i, j-1, k) / N_arr(i,j-1,k));
+                    }
+
+                // 1D
+#else
+                    amrex::Real Ux_pz = 0.0, Ux_mz = 0.0, Uy_pz = 0.0, Uy_mz = 0.0, Uz_pz = 0.0, Uz_mz = 0.0;
+                    if (N_arr(i+1,j,k) > 0.0) {
+                        Ux_pz = (NUx_arr(i+1, j, k) / N_arr(i+1,j,k));
+                        Uy_pz = (NUy_arr(i+1, j, k) / N_arr(i+1,j,k));
+                        Uz_pz = (NUz_arr(i+1, j, k) / N_arr(i+1,j,k));
+                    }
+                    if (N_arr(i-1,j,k) > 0.0) {
+                        Ux_mz = (NUx_arr(i-1, j, k) / N_arr(i-1,j,k));
+                        Uy_mz = (NUy_arr(i-1, j, k) / N_arr(i-1,j,k));
+                        Uz_mz = (NUz_arr(i-1, j, k) / N_arr(i-1,j,k));
+                    }
+#endif
+
+
+                    amrex::Real Uz_sq = Uz*Uz; amrex::Real Uy_sq = Uy*Uy; amrex::Real Ux_sq = Ux*Ux;
+                    amrex::Real c_sq = clight*clight;
+                    amrex::Real gamma = sqrt(1.0 + (Ux_sq + Uy_sq + Uz_sq)/(c_sq) );
```

Comment:
It seems that these variables are only used once, so maybe this can be simplified to:
```suggestion
                    amrex::Real c_sq = clight*clight;
                    amrex::Real gamma = sqrt(1.0 + (Ux*Ux + Uy*Uy + Uz*Uz)/(c_sq) );
```

Reply:
Change made.